pkg/cloud/api/check.go

```diff
@@ -171,3 +171,42 @@ func checkSchema(t reflect.Type) error {
 
 	return nil
 }
+
+// CheckConversion checks if we can convert type From to type To. Conversion is
+// possible if type From has the subset of fields with the same type and name as
+// in To type. This function is not symmetric. Path parameter is used for better
+// error reporting.
+func CheckConversion(p Path, from, to reflect.Type) error {
+	if from.Kind() != to.Kind() {
+		return fmt.Errorf("%s has different type: %v != %v", p.String(), from.Kind(), to.Kind())
+	}
+	switch from.Kind() {
+	case reflect.Pointer:
+		return CheckConversion(p.Pointer(), from.Elem(), to.Elem())
+
+	case reflect.Struct:
+		for i := 0; i < from.NumField(); i++ {
+			af := from.Field(i)
+			bf, exist := to.FieldByName(af.Name)
+			if !exist {
+				return fmt.Errorf("%s: type %T does not have field %v", p.String(), to, af.Name)
+			}
+			if err := CheckConversion(p.Field(af.Name), af.Type, bf.Type); err != nil {
+				return err
+			}
+		}
+		return nil
+
+	case reflect.Slice:
+		return CheckConversion(p.AnySliceIndex(), from.Elem(), to.Elem())
+
+	case reflect.Map:
+		path := p.AnyMapIndex()
+		err := CheckConversion(path, from.Key(), to.Key())
+		if err != nil {
+			return err
+		}
+		return CheckConversion(path, from.Elem(), to.Elem())
```

Comment:
we should document

We should add all of the types:

```
case reflect.Channel:
  // this is an error...
default:
  // Assume 
```

Reply:
added check for Basic type and error on not supported (others)